trin-execution/src/trie_walker/filter.rs

```diff
@@ -0,0 +1,120 @@
+use alloy::primitives::{B256, U256};
+use rand::{thread_rng, Rng};
+
+use crate::utils::partial_nibble_path_to_right_padded_b256;
+
+#[derive(Debug, Clone)]
+pub struct Filter {
+    start_prefix: U256,
```

Comment:
nit: why do we call this `_prefix`? Wouldn't `start` and `end` be sufficient and more correct?

Reply:
sure